seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/sink/writer/HdfsTransactionStateFileWriteFactory.java

```diff
@@ -98,8 +99,8 @@ public static TransactionStateFileWriter of(@NonNull SeaTunnelRowType seaTunnelR
                     rowDelimiter,
                     fileSystem);
         }
-        // if file type not supported by file connector, default txt writer will be generated
-        return new HdfsTxtTransactionStateFileWriter(
+        if (fileFormat.equals(FileFormat.EXCEL)) {
```

Comment:
In fact, you don't need to do this, we can automatically choose the format based on the type